File: languages/django_models/event.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Event(AbstractElementModel):

    # Nature
    history = models.TextField(blank=True, null=True)
    challenges = models.TextField(blank=True, null=True)
    consequences = models.TextField(blank=True, null=True)
    start_date = models.PositiveIntegerField(blank=True, null=True)
    end_date = models.PositiveIntegerField(blank=True, null=True)
    triggers = models.ManyToManyField("Event", blank=True, related_name="event_triggers")

    # Involves
    characters = models.ManyToManyField("Character", blank=True, related_name="event_characters")
    objects = models.ManyToManyField("Object", blank=True, related_name="event_objects")
    locations = models.ManyToManyField("Location", blank=True, related_name="event_locations")
    species = models.ManyToManyField("Species", blank=True, related_name="event_species")
    creatures = models.ManyToManyField("Creature", blank=True, related_name="event_creatures")
    institutions = models.ManyToManyField("Institution", blank=True, related_name="event_institutions")
    traits = models.ManyToManyField("Trait", blank=True, related_name="event_traits")
    collectives = models.ManyToManyField("Collective", blank=True, related_name="event_collectives")
    territories = models.ManyToManyField("Territory", blank=True, related_name="event_territories")
    abilities = models.ManyToManyField("Ability", blank=True, related_name="event_abilities")
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="event_phenomena")
    languages = models.ManyToManyField("Language", blank=True, related_name="event_languages")
    families = models.ManyToManyField("Family", blank=True, related_name="event_families")
    relations = models.ManyToManyField("Relation", blank=True, related_name="event_relations")
    titles = models.ManyToManyField("Title", blank=True, related_name="event_titles")
    constructs = models.ManyToManyField("Construct", blank=True, related_name="event_constructs")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Event.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Event : Element
{
    [JsonProperty("history"), TextAttribute("")]
    public string history;
    [JsonProperty("challenges"), TextAttribute("")]
    public string challenges;
    [JsonProperty("consequences"), TextAttribute("")]
    public string consequences;
    [JsonProperty("start_date"), Integer(0)]
    public int startDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("triggers"), ReferenceAttribute(typeof(Event), true)]
    public string triggers;
    [JsonProperty("characters"), ReferenceAttribute(typeof(Character), true)]
    public string characters;
    [JsonProperty("objects"), ReferenceAttribute(typeof(Object), true)]
    public string objects;
    [JsonProperty("locations"), ReferenceAttribute(typeof(Location), true)]
    public string locations;
    [JsonProperty("species"), ReferenceAttribute(typeof(Species), true)]
    public string species;
    [JsonProperty("creatures"), ReferenceAttribute(typeof(Creature), true)]
    public string creatures;
    [JsonProperty("institutions"), ReferenceAttribute(typeof(Institution), true)]
    public string institutions;
    [JsonProperty("traits"), ReferenceAttribute(typeof(Trait), true)]
    public string traits;
    [JsonProperty("collectives"), ReferenceAttribute(typeof(Collective), true)]
    public string collectives;
    [JsonProperty("zones"), ReferenceAttribute(typeof(Zone), true)]
    public string zones;
    [JsonProperty("abilities"), ReferenceAttribute(typeof(Ability), true)]
    public string abilities;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("languages"), ReferenceAttribute(typeof(Language), true)]
    public string languages;
    [JsonProperty("families"), ReferenceAttribute(typeof(Family), true)]
    public string families;
    [JsonProperty("relations"), ReferenceAttribute(typeof(Relation), true)]
    public string relations;
    [JsonProperty("titles"), ReferenceAttribute(typeof(Title), true)]
    public string titles;
    [JsonProperty("constructs"), ReferenceAttribute(typeof(Construct), true)]
    public string constructs;
}
